rotkehlchen/api/rest.py

```diff
@@ -4125,3 +4120,115 @@ def get_all_counterparties(self) -> Response:
+            addresses=addresses,
+        )
+        serialized = [entry.serialize() for entry in entries]
+        return api_response(_wrap_in_ok_result(serialized))
+
+    def add_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            entries: List[AddressbookEntry],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        try:
+            db_addressbook.add_addressbook_entries(book_type=book_type, entries=entries)
+            return api_response(result=OK_RESULT)
+        except InputError as e:
+            return api_response(
+                result=wrap_in_fail_result(str(e)),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+    def update_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            entries: List[AddressbookEntry],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        try:
+            db_addressbook.update_addressbook_entries(book_type=book_type, entries=entries)
+            return api_response(result=OK_RESULT)
+        except InputError as e:
+            return api_response(
+                result=wrap_in_fail_result(str(e)),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+    def delete_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            addresses: List[ChecksumEthAddress],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        try:
+            db_addressbook.delete_addressbook_entries(book_type=book_type, addresses=addresses)
+            return api_response(result=OK_RESULT)
+        except InputError as e:
+            return api_response(
+                result=wrap_in_fail_result(str(e)),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+    def search_for_names_everywhere(self, addresses: List[ChecksumEthAddress]) -> Response:
+        """This method searches for all names of provided addresses known to rotki. We can show
+        only one name per address, and thus we prioritize known names. Priority is the following:
+        blockchain account labels -> private addressbook -> global addressbook
+        -> ethereum tokens -> hardcoded mappings -> ENS names.
+        """
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        db_ens = DBEns(self.rotkehlchen.data.db)
+
+        reverse_ens = db_ens.get_reverse_ens(addresses=addresses).values()
+        mapped_tokens = {}
+        found_tokens = GlobalDBHandler().get_ethereum_tokens_by_addresses(addresses=addresses)
+        for token in found_tokens:
+            mapped_tokens[token.ethereum_address] = token.name
+        hardcoded_mappings = {}
+        for address in addresses:
+            constant_name = ETHADDRESS_TO_KNOWN_NAME.get(address)
+            if constant_name is not None:
+                hardcoded_mappings[address] = constant_name
```

Comment:
the hard coded names should moveto the global addressbook. Can be done later though. The global address book should be like the assets table in the global DB eventually

Reply:
I got it, will do in a separate PR in order to don't enlarge this one